doc/ufunc_examples.rst

```diff
@@ -0,0 +1,314 @@
+        v_diff_y = diff(v, axis=-1)
+        # Need to trim off elements so that the two arrays have same shape
+        div = u_diff_x[..., :-1] + v_diff_y[..., :-1, :]
+        return div
+
+Here we have treated the components of the ``(U, V)`` vector as independent scalars.
+
+Now we can compute the divergence of our example vector field
+
+.. ipython:: python
+
+    div = divergence(grid, ds["U"], ds["V"], axis=[("X", "Y"), ("X", "Y")])
+
+We can see the result lies on the expected coordinate positions
+
+.. ipython:: python
+
+    div.coords
+
+and the resulting divergence looks like it corresponds with the arrows of the vector field above
+
+.. ipython:: python
+
+    import matplotlib.pyplot as plt
+
+    div.plot(x="x_c", y="y_c")
+    colocated.plot.quiver("x_c", "y_c", u="U", v="V")
+
+    @savefig div_vector_field.png width=4in
+    plt.gcf()
+
+
+
+Gradient
+~~~~~~~~
+
+The gradient is almost like the opposite of divergence in the sense that it accepts one scalar and returns multiple vectors.
+
+Let's first define a tracer field ``T``, which we imagine will start off localised near the center of the domain.
+
+.. ipython:: python
+
+    def gaussian(x_coord, y_coord, x_pos, y_pos, A, w):
+        return A * np.exp(
+            -0.5 * ((x_coord - x_pos) ** 2 + (y_coord - y_pos) ** 2) / w ** 2
+        )
+
+    ds["T"] = gaussian(grid_ds.x_c, grid_ds.y_c, x_pos=7.5, y_pos=7.5, A=50, w=2)
+
+    @savefig tracer_field.png width=4in
+    ds["T"].plot.contourf(x="x_c", vmax=60)
+
+Computing the first-order gradient will again move the data onto different grid positions,
+so the signature for a gradient ufunc will need to reflect this
+and our definition is similar to the derivative case.
+
+.. ipython:: python
+
+    def gradient(a):
+        a_diff_x = diff(a, axis=-2)
+        a_diff_y = diff(a, axis=-1)
+        # Need to trim off elements so that the two arrays have same shape
+        return a_diff_x[..., :-1], a_diff_y[..., :-1, :]
+
+Now we can compute the gradient of our example scalar field
+
+.. ipython:: python
+
+    ds["grad_T_x"], ds["grad_T_y"] = grid.apply_as_grid_ufunc(
+        gradient,
+        ds["T"],
+        axis=[("X", "Y")],
+        signature="(X:center,Y:center)->(X:left,Y:center),(X:center,Y:left)",
+        boundary_width={"X": (1, 0), "Y": (1, 0)},
+    )
+
+.. note::
+
+    Notice we used the ``apply_as_grid_ufunc`` syntax here instead of the ``as_grid_ufunc`` decorator.
+    The result is the same.
+
+Again in order to plot this as a vector field we should first interpolate it
+
+.. ipython:: python
+
+    colocated["grad_T_x"] = grid.interp(ds["grad_T_x"], axis="X", to="center")
+    colocated["grad_T_y"] = grid.interp(ds["grad_T_y"], axis="Y", to="center")
+    colocated
+
+Now we can plot the gradient of the tracer field as a vector field
+
+.. ipython:: python
+
+
+    ds["T"].plot.contourf(x="x_c", vmax=60)
+    colocated.plot.quiver("x_c", "y_c", u="grad_T_x", v="grad_T_y", scale=200)
```

Comment:
```suggestion
    colocated.plot.quiver("x_c", "y_c", u="grad_T_x", v="grad_T_y", c='0.5', scale=200)
```

Reply:
`AttributeError: 'Quiver' object has no property 'c'`

What is this supposed to do?